libbpfgo.go

```diff
@@ -1423,8 +1412,8 @@ func (hook *TcHook) Detach(tcOpts *TcOpts) error {
 func (hook *TcHook) Query(tcOpts *TcOpts) error {
 	opts := tcOptsToC(tcOpts)
 	ret := C.bpf_tc_query(hook.hook, opts)
-	if ret < 0 {
-		return syscall.Errno(-ret)
+	if ret != 0 {
+		return syscall.Errno(ret)
```

Comment:
Errors are returned negative. See https://github.com/libbpf/libbpf/blob/e99f34e14447866bcbc38944387531c36f6c85fe/src/libbpf_internal.h#L454

And https://github.com/libbpf/libbpf/blob/21f90f61b0849ae654b7c78ba9ce34bfb74ce6f2/src/netlink.c#L756

Reply:
The discussion about this was made at:

https://github.com/aquasecurity/libbpfgo/pull/100#discussion_r771589769

I wonder if I missed something there.

Reply:
> You call bpf_object__load(NULL)... then you will face:
> 
> ```c
> if (!attr)
>         return libbpf_err(-EINVAL);
> ```
> 
> and `libbpf_err(-EINVAL):`:
> 
> ```c
> static inline int libbpf_err(int ret)
> {
>         if (ret < 0)
>                 errno = -ret;
>         return ret;
> }
> ```
> 
> So it will return a negative number, but errno will still be positive on the error return. Why do you need to set it to negative again ?

Like you write here, the return is a negative number, meaning ret is negative. To have a positive value, we should `syscall.Errno(-ret)`